File: crates/store/re_types/src/datatypes/utf8pair_ext.rs
use crate::datatypes::Utf8;

impl<T: Into<Utf8>> From<(T, T)> for super::Utf8Pair {
    #[inline(always)]
    fn from(value: (T, T)) -> Self {
        Self {
            first: value.0.into(),
            second: value.1.into(),
        }
    }
}

impl From<&(&str, &str)> for super::Utf8Pair {
    #[inline(always)]
    fn from(value: &(&str, &str)) -> Self {
        Self {
            first: value.0.into(),
            second: value.1.into(),
        }
    }
}

File: rerun_cpp/src/rerun/datatypes/utf8pair_ext.cpp
#include <utility>
#include "utf8pair.hpp"

// #define EDIT_EXTENSION

namespace rerun {
    namespace datatypes {

#ifdef EDIT_EXTENSION
        // <CODEGEN_COPY_TO_HEADER>

        /// Creates a string pair.
        Utf8Pair(rerun::datatypes::Utf8 first_, rerun::datatypes::Utf8 second_)
            : first(std::move(first_)), second(std::move(second_)) {}

        // </CODEGEN_COPY_TO_HEADER>
#endif
    } // namespace datatypes
} // namespace rerun
